os/board/rtl8721csm/src/component/common/bluetooth/realtek/sdk/example/ble_peripheral/ble_tizenrt_server.c

```diff
@@ -370,24 +362,37 @@ trble_result_e rtw_ble_server_set_adv_data(uint8_t* data, uint16_t length)
     }
 
     do {   
-        debug_print("\r\n[%s] Waiting for adv idle , now %d", __FUNCTION__, new_state.gap_adv_state);
+        debug_print("\r\n[%s] Waiting for adv idle", __FUNCTION__);
         os_delay(100);
         le_get_gap_param(GAP_PARAM_DEV_STATE , &new_state);
     } while(new_state.gap_adv_state != GAP_ADV_STATE_IDLE);
 
     debug_print("\r\n[%s] ADV STATE : IDLE", __FUNCTION__);
+}
+
+/* Set Advertisement Data API */
+trble_result_e rtw_ble_server_set_adv_data(uint8_t* data, uint16_t length)
+{
+    if (is_server_init != true)
+        return TRBLE_INVALID_STATE;
+
+    if(length > 31)
+        return TRBLE_FAIL;
```

Comment:
How about returning `TRBLE_INVALID_ARGS` instead of `TRBLE_FAIL`?

Reply:
Done.